src/wp-admin/includes/class-wp-filesystem-ftpsockets.php

```diff
@@ -414,19 +414,17 @@ public function delete( $file, $recursive = false, $type = false ) {
 	 * Checks if a file or directory exists.
 	 *
 	 * @since 2.5.0
+	 * @since 6.0.0 Use file size.
 	 *
 	 * @param string $file Path to file or directory.
 	 * @return bool Whether $file exists or not.
 	 */
 	public function exists( $file ) {
-		$list = $this->ftp->nlist( $file );
-
-		if ( empty( $list ) && $this->is_dir( $file ) ) {
+		if ( $this->is_dir( $file ) ) {
 			return true; // File is an empty directory.
 		}
 
-		return ! empty( $list ); // Empty list = no file, so invert.
-		// Return $this->ftp->is_exists($file); has issues with ABOR+426 responses on the ncFTPd server.
+		return 0 > $this->size( $file );
```

Comment:
Should we use `0 > $this->size( $file )` or `-1 !== $this->size( $file )`? This would make the method similar to the change in `FTPExt::exists()`.

Reply:
In FTPsockets the size() uses `$this->ftp->file size()` I don't really know it's return values as I don't know what function it references.

Reply:
I think the update that @afragen just made (`return false !== $size && is_numeric( $size );`) is the right way to do it

Reply:
Yep, we just simplified it down to `is_numeric()` since `false` isn't numeric anyway. This looks good to me!

Reply:
I had to double check that `is_numeric( false )` didn't return `true` (which it does **not**)